R/ci.R

```diff
@@ -147,3 +147,66 @@ check_uses_gitlab_ci <- function(base_path = proj_get()) {
     "
   )
 }
+
+#' @section `use_circleci()`:
+#' Adds a basic `.circleci/config.yml` to the top-level directory of a package. This is a
+#' configuration file for the [CircleCI](https://circleci.com/) continuous
+#' integration service.
+#' @param image The Docker image to use for build. Must be available on
+#'   [DockerHub](https://hub.docker.com). The
+#'   [rocker/verse](https://hub.docker.com/r/rocker/verse) image includes TeX
+#'   Live, pandoc, and the tidyverse packages. For a minimal image, try
+#'   [rocker/r-ver](https://hub.docker.com/r/rocker/r-ver). To specify a version
+#'   of R, change the tag from `latest` to the version you want, e.g.
+#'   `rocker/r-ver:3.5.3`.
+#' @export
+#' @rdname ci
+use_circleci <- function(browse = interactive(), image = "rocker/verse:latest") {
+  check_uses_github()
+  use_directory(".circleci")
+  new <- use_template(
+    "circleci-config.yml",
+    ".circleci/config.yml",
+    data = list(package = project_name(), image = image),
+    ignore = TRUE
+  )
+  if (!new) return(invisible(FALSE))
```

Comment:
Can you run styler over this? I think we want this in curly braces, with newlines.